Directory.Build.targets

```diff
@@ -147,7 +147,7 @@
     <KnownFrameworkReference Condition="'$(UseAspNetCoreSharedRuntime)' != 'true'" Remove="Microsoft.AspNetCore.App" />
     <KnownFrameworkReference Remove="Microsoft.WindowsDesktop.App" />
 
-    <KnownFrameworkReference Condition="'$(UseAspNetCoreSharedRuntime)' == 'true'" Update="Microsoft.AspNetCore.App">
+    <KnownFrameworkReference Condition="'$(UseAspNetCoreSharedRuntime)' == 'true' AND '$(DoNotApplyWorkaroundsToMicrosoftAspNetCoreApp)' != 'true'" Update="Microsoft.AspNetCore.App">
```

Comment:
FYI @wtgodbe

Reply:
Remind me what the motivation was for this?

Reply:
We wanted a middleware that targeted netcoreapp3.1 and compiled against M.AspNetCore.App. `UseAspNetCoreSharedRuntime` is required to keep the shared runtime reference around, but I didn't want to compile against 5.0.0-dev.